integrationTest/integration_test.go

```diff
@@ -997,6 +999,50 @@ func makeTests(t *testing.T) []*TestGroup {
 			),
 		),
 
+		testgroup("DS (children only) CLOUDNS",
+			requires(providers.CanUseDSForChildren),
+			only("CLOUDNS"),
+			// Use a valid digest value here, because GCLOUD (which implements this capability) verifies
+			// the value passed in is a valid digest. RFC 4034, s5.1.4 specifies SHA1 as the only digest
+			// algo at present, i.e. only hexadecimal values currently usable.
+			// Cloudns requires NS  Record before creating DS Record.
+			tc("create DS",
+				ns("child", "ns101.cloudns.net."),
+				ds("child", 35632, 13, 1, "1E07663FF507A40874B8605463DD41DE482079D6"),
```

Comment:
Does this test work if we list `ds(` before `ns(`?  The provider should do the right thing in either case.

Reply:
- NS record must be added before adding DS record
- DS record must be deleted before we can delete NS record

Reply:
> Does this test work if we list `ds(` before `ns(`? The provider should do the right thing in either case.

Any hint how we can get the provider to "order" the entries in the diff?
As we are new in this, we did not find the right place to do this.

Because when deleting entries, DS needs to be deleted before NS, too.

Reply:
Other providers make multiple passes over the records.  For example, the first pass only does DS records, the second pass does all records. 

Alternatively just move the DS records to the top of the list so they happen first.

Reply:
with the last changes, DS record can be created even if we list `ds(` before `ns(`. The test work